src/libmongoc/tests/json-test-monitoring.c

```diff
@@ -520,16 +543,20 @@ check_json_apm_events (json_test_ctx_t *ctx, const bson_t *expectations)
             /* if we allow matching only a subset of actual events, skip
              * non-matching ones */
             continue;
-         } else {
-            _apm_match_error_context (&ctx->events, expectations);
-            test_error ("could not match APM event\n"
-                        "\texpected: %s\n\n"
-                        "\tactual  : %s\n\n"
-                        "\terror   : %s\n\n",
-                        bson_as_canonical_extended_json (&expectation, NULL),
-                        bson_as_canonical_extended_json (&actual, NULL),
-                        match_ctx.errmsg);
          }
+
+         if (skip_cse_list_collections (&actual)) {
```

Comment:
This skip logic is specific to CSFLE tests, but `check_json_apm_events` applies to all spec tests that exercise this code, (CRUD, transactions).

There's a very small risk a non CSFLE test could skip an event if there is a legitimate `listCollections` against a `keyvault` database in a spec test.

I'm not too concerned though. It seems very unlikely. And it would result in a test failure (not a loss of test coverage). If it becomes an issue, we can make this configurable later.